### ISSUE TITLE
'resnet 110' as teacher, 'resnet20' as TA, 'resnet8' as student on CIFAR100’

### ISSUE THREAD

**Issue by InstantWindy** [imirzadeh/Teacher-Assistant-Knowledge-Distillation#1]
Hi! What's meaning of the ‘resnet20 as Ta’? why not directely treat the resnet 110 as teacher and treat the resnet8 as student?  why adding a resnet20 as Ta. looking forward to your reply!Thanks

**Comment by imirzadeh** [comment]
Hi!
The details are available in the paper: [https://arxiv.org/abs/1902.03393](https://arxiv.org/abs/1902.03393)
However, I can explain the intuition:   
The reason is that resnet110 is a too powerful teacher for resnet8 as a student. But, resnet110 is not a too powerful teacher for resnet20 as a student. Also, resnet20 is not a too powerful teacher for resnet8. Roughly it means, resnet8 can understand resnet20 well and resnet20 can understand resnet110 well but resnet8 cannot understand resnet110 well. So, we use resnet20 as a proxy to distillate knowledge from resnet110 to resnet8.

**Comment by InstantWindy** [comment]
Thanks, I got it. I am not clear that what the  parameter T does, could  you explain the meaning of T？ Thanks!

**Comment by imirzadeh** [comment]
Parameter 'T' refers to the temperature in equation(2) of the paper. It also explained in equation(1) of Hinton's paper on knowledge distillation [https://arxiv.org/pdf/1503.02531.pdf](https://arxiv.org/pdf/1503.02531.pdf).    
Roughly it means how soft you want to make the teacher's logits you are using for distillation. The more the temperature, the softer the logits.

**Comment by InstantWindy** [comment]
But I have a question, what's the benefit of the softer the logits? I am a bit confused. Thanks!

**Comment by imirzadeh** [comment]
It's easier to learn from soft logits than hard logits. you can check other papers on knowledge distillation where they discuss the details.
